example/src/screens/ClientListScreen.tsx

```diff
@@ -2,174 +2,92 @@
+import { ButtonGroup } from "react-native-elements";
 
-type WebSocketClientProps = { name: string; client: Client };
-const WebSocketClient = ({ name, client }: WebSocketClientProps) => (
-    <View style={styles.row}>
-        <View style={styles.clientUrl}>
-            <Text>{name}</Text>
-            <Text>{client.wsUrl}</Text>
-        </View>
-        <View style={styles.clientAction}>
-            <Text>{">"}</Text>
-        </View>
-    </View>
-);
+import GenericClient from "@mattermost/react-native-network-client";
 
-const ItemSeparator = () => <View style={styles.separator} />;
+import ClientListItem from "../components/ClientListItem";
+import { createMattermostAPIClient } from "../utils";
 
 export default function ClientListScreen({
     navigation,
     route,
 }: ClientListScreenProps) {
-    const [clients, setClients] = useState<Client[]>([
-        { type: "network", client: GenericClient, name: "Generic Client" },
+    const [clients, setClients] = useState<NetworkClientItem[]>([
+        { client: GenericClient, name: "Generic Client" },
     ]);
 
     useEffect(() => {
-        if (route.params?.client) {
-            setClients([...clients, route.params.client]);
+        if (clients.length === 1) {
+            createMattermostAPIClient().then((client) => {
+                if (client) {
+                    setClients([...clients, client]);
+                }
+            });
+        }
+    }, []);
+
+    useEffect(() => {
+        if (route.params?.createdClient) {
+            setClients([...clients, route.params.createdClient]);
         }
-    }, [route.params?.client]);
+    }, [route.params?.createdClient]);
 
     const deleteClient = (clientIndex: number) => {
         setClients(clients.filter((_, index) => clientIndex !== index));
     };
 
-    const renderItem = ({ item, index }: { item: Client; index: number }) =>
-        item.type === "network" ? (
-            <NetworkClient
-                name={item.name}
-                index={index}
-                client={item.client}
-                deleteClient={deleteClient}
-                navigate={navigation.navigate}
-            />
-        ) : (
-            <WebSocketClient name={item.name} client={item} />
-        );
+    const renderItem = ({
+        item,
+        index,
+    }: {
+        item: NetworkClientItem;
+        index: number;
+    }) => (
+        <ClientListItem
+            name={item.name}
+            index={index}
+            client={item.client}
+            deleteClient={deleteClient}
+            navigate={navigation.navigate}
+        />
+    );
 
-    const keyExtractor = (item: Client) =>
-        item.baseUrl || item.wsUrl || item.name;
+    const keyExtractor = (item: NetworkClientItem) => {
+        if ("baseUrl" in item.client) {
+            return item.client.baseUrl;
+        } else if ("wsUrl" in item.client) {
+            return item.client.wsUrl;
+        }
+
+        return item.name;
+    };
```

Comment:
This might also be good to have in the utils file rather than just the client screen one.